### PR TITLE
V4.0 iometrics

### PR DESCRIPTION
**Pull request by aviviadi**
http://issues.hibernatingrhinos.com/issue/RavenDB-4833

### CODE REVIEW

**Comment by redknightlois** [comment]
Are we going to adjust automatically to IO rate if we compress less than 64Kb writes?

**Comment by ayende** [comment]
@redknightlois No, we have tested that, and it didn't provide any meaningful difference for small writes on fast / slow / network disks

**Comment by ayende** [comment]
The tests are failing

File: src/Voron/Impl/Journal/WriteAheadJournal.cs [hunk 1]
```diff
@@ -928,6 +937,11 @@ private CompressedPagesResult CompressPages(LowLevelTransaction tx, int numberOf
             byte* tempBuffer = null;
             byte* write;
             bool isSizeBiggerThan64K = sizeInBytes + pageSize > 64*1024;  // +pageSize to count TransactionHeader
+
+            // even when it seems not worth to compress, if i/o is too slow - we do benefit from compression
+            if (isSizeBiggerThan64K == false && _isSlowIO)
+                    isSizeBiggerThan64K = true;
```
Comment (ayende) [inline src/Voron/Impl/Journal/WriteAheadJournal.cs:943]: This is misleading name

**Pull request closed without merging**